src/index.ts

```diff
@@ -105,16 +154,17 @@ export function stakingTransaction(
     value: amount,
   });
 
-  if (dataEmbedScript) {
+  if (scripts.dataEmbedScript) {
     // Add the data embed output to the transaction
     psbt.addOutput({
-      script: dataEmbedScript,
+      script: scripts.dataEmbedScript,
       value: 0,
     });
   }
 
   // Add a change output only if there's any amount leftover from the inputs
-  if (inputsSum > amount + fee) {
+  const inputsSum = inputValueSum(selectedUTXOs);
+  if ((inputsSum + BTC_DUST_SAT) > (amount + fee)) {
```

Comment:
```suggestion
  if (inputsSum - (amount + fee) < BTC_DUST_SAT) {
```

We do not want `inputsSum - (amount + fee) < BTC_DUST_SAT`, i.e. to not add an additional output which contains below the number Bitcoin considers as dust.

For example, if `inputsSum = 100`, `amount = 90`, `fee=8`, and `BTC_DUST_SAT=10`, then we should not add a change output, as it will contain only `2` which is below the dust output. The above condition will satisfy this though. cc @KonradStaniec

Reply:
the if condition is to "add the change output".
so the condition should be `if (inputsSum - (amount + fee) > BTC_DUST_SAT)`